hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ServicesState.java

```diff
@@ -396,6 +396,16 @@ private ServicesApp internalInit(
             app.initializationFlow().runWith(this, bootstrapProps);
             if (trigger == RESTART && isUpgrade) {
                 app.stakeStartupHelper().doUpgradeHousekeeping(networkCtx(), accounts(), stakingInfo());
+                // FIXME: uncomment after migration to plaftorm-sdk 0.38+,
+                // https://github.com/hashgraph/hedera-services/issues/6037
```

Comment:
This may not be needed. If this PR is merged before accounts/tokens/etc. are moved from MerkleMap to VirtualMap, there will be no compatibility issues.

Reply:
If the tree is in the correct state, it's going to be no-op anyway, and for the virtual maps that are already in use it's absolutely necessary. If I remember correctly, I put this code here because `VirtualMap`-s are already in use in case of `ACCOUNTS` and `TOKEN_ASSOCIATIONS`. Is my assessment incorrect?

Reply:
After a conversation with Artem we decided to keep things the way they are. However, I updated the comment to specify more accurately when the change has to happen (namely, immediately after this PR is merged rather than migration to platform-sdk 0.38)